ofproto/ofproto.c

```diff
@@ -5606,11 +5605,24 @@ handle_update_map(struct ofconn *ofconn, const struct ofp_header *oh)
         return OFPERR_OFPBRC_EPERM;
     }
 
-    if (ubpf_map_update(map, key, value) < 0) {
-        VLOG_WARN_RL(&rl,
-                     "The update map operation has failed.");
-        return OFPERR_OFPBRC_EPERM;
+    void *key = malloc(msg.key_size);
+    void *value = malloc(msg.value_size);
+    for(int i = 0; i < msg.nb_elems; i++) {
+        memcpy(key, data, msg.key_size);
+        data += msg.key_size;
+        memcpy(value, data, msg.value_size);
+        data += msg.value_size;
```

Comment:
Do we need these `memcpy`s? What about:
```c
void *key = data;
data += msg.key_size;
void *value = data;
data += msg.value_size;
if (ubpf_map_update(map, key, value) < 0) {
...
```
`ubpf_map_update` should only consume `key_size` and `value_size` bytes anyway.